File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    var answer = [];
    var min = Math.min(...arr);
    arr.forEach((element) => {
        if (element != min){
            answer.push(element);
        }
    });
    console.log(arr);
    if (answer.length == 0) answer.push(-1);
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int minNum = Integer.MAX_VALUE;
        List<Integer> list = new ArrayList<>();
        
        for (int i = 0; i < arr.length; i++) {
            minNum = Math.min(arr[i], minNum);
        }
        
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] != minNum) list.add(arr[i]);
        }
        
        if (list.isEmpty()) {
            return new int[]{-1};
        } else {
            int[] result = new int[list.size()];
            for (int i = 0; i < list.size(); i++) {
                result[i] = list.get(i);
            }
            return result;
        }
    }
}